### DISCUSSION TITLE
How to disable credentials from environment variables

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#12118] [Request Help] [answered]
Today I learned that Renovate supports credentials set via environment variables. Looking at https://github.com/renovatebot/renovate/blob/main/lib/workers/global/config/parse/env.ts#L123-L173 I understand that it looks for environment variables the start with the name of a known data source. When it discovers such an variable it expects a certain structure in the variable name and prints a warning when that structure is not fulfilled.

In our set up (self-hosted on our internal GitLab) this leads to the following message in our logs:

![image](https://user-images.githubusercontent.com/1540469/136812213-c3c5ad5c-4dea-4a70-a776-a653c9aab467.png)

This variable is configured centrally for other purposes and cannot be easily renamed.

Is there a way to disable this behavior?

**Comment by viceice** [comment]
I suggested a flag when this was implemented, but  that was denied. 🤷‍♂️

**Comment by rarkins** [comment, marked as answer]
No, but please create a feature request so we can discuss. Maybe we add a pattern for which to match, defaulting it to `.*`?

**Reply by fgreinacher** [reply]
Sure: #12155
